kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisSource.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.kinesis.scaladsl
+
+import akka.NotUsed
+import akka.stream.alpakka.kinesis.KinesisSourceErrors.NoShardsError
+import akka.stream.alpakka.kinesis.{KinesisSourceStage, ShardSettings}
+import akka.stream.scaladsl.{Merge, Source}
+import com.amazonaws.services.kinesis.AmazonKinesisAsync
+import com.amazonaws.services.kinesis.model.{Record, Shard, StreamDescription}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object KinesisSource {
+
+  def basic(shardSettings: ShardSettings, amazonKinesisAsync: AmazonKinesisAsync): Source[Record, NotUsed] =
+    Source.fromGraph(new KinesisSourceStage(shardSettings, amazonKinesisAsync))
+
+  def basicMerge(shardSettings: List[ShardSettings], amazonKinesisAsync: AmazonKinesisAsync): Source[Record, NotUsed] = {
+    val create: ShardSettings => Source[Record, NotUsed] = basic(_, amazonKinesisAsync)
+    shardSettings match {
+      case Nil => Source.failed(NoShardsError)
+      case first :: Nil => create(first)
+      case first :: second :: Nil => Source.combine(create(first), create(second))(Merge(_))
+      case first :: second :: rest =>
+        Source.combine(create(first), create(second), rest.map(create(_)): _*)(Merge(_))
+    }
+  }
+
+  def getShardsByStreamName(streamName: String, amazonKinesisAsync: AmazonKinesisAsync): Try[List[Shard]] = {
+    @tailrec
+    def getShards(streamDescription: StreamDescription, shards: java.util.ArrayList[Shard]): List[Shard] =
+      if (streamDescription.getHasMoreShards) {
+        shards.addAll(streamDescription.getShards)
+        getShards(streamDescription, shards)
+      } else shards.asScala.toList
+
+    Try(amazonKinesisAsync.describeStream(streamName).getStreamDescription)
```

Comment:
We should call the async variant of describeStream here and return a scala future. Also this method is missing in the javadsl.KinesisSource object. Therefore it makes sense to move the implementation that returns java future to the javadsl package and change the implementation here to call to javadsl and then convert to scala future.